### ISSUE TITLE
Add msg.respond()

### ISSUE THREAD

**Issue by wallyqs** [nats-io/nats.py#126]
Similar to the Go client.

**Comment by Bernolt** [comment]
I think I could implement this and the test associated..

I would add it as a method of class Msg.
It would require to add a reference to Client within Msg in order to call Client.publish().

Seems ok for you?
```
// Respond allows a convenient way to respond to requests in service based subscriptions.                                                                                                                   
func (m *Msg) Respond(data []byte) error {
        if m == nil || m.Sub == nil {
                return ErrMsgNotBound
        }
        if m.Reply == "" {
                return ErrMsgNoReply
        }
        m.Sub.mu.Lock()
        nc := m.Sub.conn
        m.Sub.mu.Unlock() to collect nc from Msg. 
        // No need to check the connection here since the call to publish will do all the checking.                                                                                                         
        return nc.Publish(m.Reply, data)
}
```

**Comment by wallyqs** [comment]
Thanks, yes I think would be fine to add that way, the Go client does the same.

**Comment by wallyqs** [comment]
Fixed via #182
